cg/meta/archive/archive.py

```diff
@@ -83,10 +99,111 @@ def archive_all_non_archived_spring_files(
+                archive_location=archive_location,
             )
 
+    def retrieve_samples(self, sample_internal_ids: List[str]) -> None:
+        """Retrieves the archived spring files for a list of samples."""
+        samples: List[Sample] = [
+            self.status_db.get_sample_by_internal_id(sample_internal_id)
+            for sample_internal_id in sample_internal_ids
+        ]
+        samples_and_destinations: List[
+            SampleAndDestination
+        ] = self.add_housekeeper_destinations_to_samples(samples)
+        for archive_location in ArchiveLocations:
+            filtered_samples: List[SampleAndDestination] = filter_samples_on_archive_location(
+                samples_and_destinations=samples_and_destinations,
+                archive_location=archive_location,
+            )
+            if filtered_samples:
+                job_id: int = self.retrieve_samples_from_archive_location(
+                    samples_and_destinations=filtered_samples,
+                    archive_location=archive_location,
+                )
+                self.set_archive_retrieval_task_ids(
+                    retrieval_task_id=job_id,
+                    files=self.get_spring_files_from_samples(
+                        [sample.sample for sample in filtered_samples]
+                    ),
+                )
+
+    def get_spring_files_from_samples(self, samples: List[Sample]) -> List[File]:
+        """Gets all spring files in Housekeeper in the bundles corresponding to the given list of samples."""
+        files: List[File] = []
+        for sample in samples:
+            files.extend(
+                self.housekeeper_api.get_files_from_latest_version(
+                    bundle_name=sample.internal_id, tags=[SequencingFileTag.SPRING]
+                ).all()
+            )
+        return files
+
+    def add_housekeeper_destinations_to_samples(
+        self, samples: List[Sample]
+    ) -> List[SampleAndDestination]:
+        """Gets all samples which were run on the specified flowcell and clumps it with the destination in
+        Housekeeper where we want the retrieved files to be."""
+        samples_to_retrieve: List[SampleAndDestination] = []
+        for sample in samples:
+            LOG.info(f"Will try to retrieve sample: {sample.internal_id}.")
+            destination: str = self.get_destination_from_sample_internal_id(sample.internal_id)
+            samples_to_retrieve.append(SampleAndDestination(sample=sample, destination=destination))
+        return samples_to_retrieve
+
+    def retrieve_samples_from_archive_location(
+        self,
+        samples_and_destinations: List[SampleAndDestination],
+        archive_location: ArchiveLocations,
+    ):
+        archive_handler: ArchiveHandler = ARCHIVE_HANDLERS[archive_location](self.data_flow_config)
+        return archive_handler.retrieve_samples(samples_and_destinations)
+
+    def retrieve_sample(self, sample_internal_id: str) -> None:
```

Comment:
Do you think we need this at the moment? I'm guessing the functionality is the same as using `retrieve_samples` with `[sample_id]` as argument?